### PR TITLE
RNIS v1 backend implementation

### PR DESCRIPTION
**Pull request by pastorsx**
Implementation of RNIS v1 backend including:
1) model change to support subtypes of POA and domain 3gpp related information
2) API subscription list GET support
3) API subscription cell_change list GET support
4) API subscription for cell_change (GET, POST, PUT, DELETE) support
5) Notification support for cell_change
6) Notification support for subscription expiry
7) API PLMNInfo GET support
8) All other API endpoint returning "Not Implemented" http error code
9) /api available through AdvantEDGE
10) integration with latest develop branch (support for ingress)

Testing:
manual system testing using http-echo server to check notifications content

### CODE REVIEW

File: go-apps/meep-ctrl-engine/api/swagger.yaml [hunk 1]
```diff
@@ -671,12 +671,31 @@ definitions:
         description: "Key/Value Pair Map (string, string)"
         additionalProperties:
           type: "string"
+      3gpp:
```
Comment (dilallkx) [inline go-apps/meep-ctrl-engine/api/swagger.yaml:674]: Rename to "cellular4gConfig"

File: go-apps/meep-ctrl-engine/api/swagger.yaml [hunk 2]
```diff
@@ -671,12 +671,31 @@ definitions:
...
+        $ref: "#/definitions/Domain3gpp"
       zones:
         type: "array"
         items:
           $ref: "#/definitions/Zone"
     description: "Operator domain object"
     example: {}
+  Domain3gpp:
```
Comment (dilallkx) [inline go-apps/meep-ctrl-engine/api/swagger.yaml:682]: Rename to "Cellular4gDomainConfig"

File: go-apps/meep-rnis/server/rnis.go [hunk 3]
```diff
@@ -0,0 +1,694 @@
...
+	for _, meAppName := range appInsIdArray {
+		meAppName = strings.TrimSpace(meAppName)
+
+		n := activeModel.GetNode(meAppName)
+		if n == nil {
+			log.Error("Error node not found " + meAppName)
+			continue
+		}
+		meApp, ok := n.(*ceModel.Process)
+		if ok {
```
Comment (dilallkx) [inline go-apps/meep-rnis/server/rnis.go:561]: To help reduce indentation:

if meApp, ok := n.(*ceModel.Process); ok {

}

Do this at each level.

File: go-packages/meep-ctrl-engine-client/api/swagger.yaml [hunk 4]
```diff
@@ -671,12 +671,31 @@ definitions:
         description: "Key/Value Pair Map (string, string)"
         additionalProperties:
           type: "string"
+      3gpp:
```
Comment (dilallkx) [inline go-packages/meep-ctrl-engine-client/api/swagger.yaml:674]: Regenerate code after data model updates suggested earlier.

File: go-packages/meep-loc-serv-notification-client/go.mod [hunk 5]
```diff
@@ -1,4 +1,4 @@
-module github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-loc-serv-notification-client
+module github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-rnis-notification-client
```
Comment (dilallkx) [inline go-packages/meep-loc-serv-notification-client/go.mod:1]: Why is this RNIS?

File: go-apps/meep-ctrl-engine/api/swagger.yaml (hunk 2 above)
Comment (pastorsx) [reply go-apps/meep-ctrl-engine/api/swagger.yaml:682]: done

File: go-apps/meep-ctrl-engine/api/swagger.yaml (hunk 1 above)
Comment (pastorsx) [reply go-apps/meep-ctrl-engine/api/swagger.yaml:674]: done

File: go-apps/meep-ctrl-engine/api/swagger.yaml [hunk 6]
```diff
@@ -808,6 +833,10 @@ definitions:
         description: "Key/Value Pair Map (string, string)"
         additionalProperties:
           type: "string"
+      cellId:
+        type: "string"
+        description: "The E-UTRAN Cell Identity as defined in ETSI TS 136 413 including\
+          \ the ID of the eNB serving the cell"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-ctrl-engine/api/swagger.yaml:839]: done

File: go-apps/meep-ctrl-engine/api/swagger.yaml [hunk 7]
```diff
@@ -782,6 +801,12 @@ definitions:
         enum:
         - "POA"
         - "DEFAULT"
+      subType:
+        type: "string"
+        description: "Network location subtype"
+        enum:
+        - "3GPP"
+        - "NON-3GPP"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-ctrl-engine/api/swagger.yaml:809]: done

File: go-apps/meep-loc-serv/server/loc-serv.go [hunk 8]
```diff
@@ -778,6 +778,7 @@ func userTrackingSubPost(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	log.Info("SIMON2 ", userTrackingSub)
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-loc-serv/server/loc-serv.go:781]: done

File: go-apps/meep-rnis/main.go [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+		sigchan := make(chan os.Signal, 10)
+		signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM)
+		<-sigchan
+		log.Info("Program killed !")
+		// do last actions and wait for all write operations to end
+		run = false
+	}()
+
+	go func() {
+		// Initialize Location Service
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-rnis/main.go:53]: done

File: go-packages/meep-ctrl-engine-client/api/swagger.yaml (hunk 4 above)
Comment (pastorsx) [reply go-packages/meep-ctrl-engine-client/api/swagger.yaml:674]: done

File: go-packages/meep-loc-serv-notification-client/go.mod (hunk 5 above)
Comment (pastorsx) [reply go-packages/meep-loc-serv-notification-client/go.mod:1]: content of rnis and loc-serv file were reversed.....?..?

fixed

File: go-apps/meep-rnis/sbi/rnis-sbi.go [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+		return err
+	}
+
+	sbi.updateUeEcgiInfoCB = updateUeEcgiInfo
+	return nil
+}
+
+func GetActiveModel() (model *mod.Model) {
+	return sbi.activeModel
+}
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-rnis/sbi/rnis-sbi.go:56]: removed. The plmnGetInfo was reworked so it uses a callback through the sbi to populate the infor required to reply to the PLMNINFO GET. The same was was used as how the ue are populated with an Ecgi. Should have been done in the first place. Now App behaves the same.

**Pull request merged**